tests/components/yale/test_config_flow.py

```diff
@@ -1,403 +1,205 @@
-"""Test the Yale config flow."""
+"""Test the yale config flow."""
 
 from unittest.mock import patch
 
-from yalexs.authenticator import ValidationResult
-from yalexs.manager.exceptions import CannotConnect, InvalidAuth, RequireValidation
+import pytest
 
 from homeassistant import config_entries
-from homeassistant.components.yale.const import (
-    CONF_ACCESS_TOKEN_CACHE_FILE,
-    CONF_BRAND,
-    CONF_INSTALL_ID,
-    CONF_LOGIN_METHOD,
-    DOMAIN,
-    VERIFICATION_CODE_KEY,
+from homeassistant.components.yale.application_credentials import (
+    OAUTH2_AUTHORIZE,
+    OAUTH2_TOKEN,
 )
-from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.components.yale.const import DOMAIN
 from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .mocks import USER_ID
 
 from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+from tests.typing import ClientSessionGenerator
 
+CLIENT_ID = "1"
 
-async def test_form(hass: HomeAssistant) -> None:
-    """Test we get the form."""
 
+@pytest.mark.usefixtures("client_credentials")
+@pytest.mark.usefixtures("current_request_with_host")
+async def test_full_flow(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+    jwt: str,
+) -> None:
+    """Check full flow."""
     result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
+        "yale", context={"source": config_entries.SOURCE_USER}
```

Comment:
Why don't we keep using domain?

Reply:
copy-pasta: I rewrote all the tests and copied from other oauth integrations.

Reply:
932538eefa3edb4ae934c989dbe5e566b3bd7040